### PR TITLE
DOC: Updated overlay docstring

### PR DESCRIPTION
**Pull request by rwijtvliet**
to reflect recent added functionality

### CODE REVIEW

File: geopandas/tools/clip.py [hunk 1]
```diff
@@ -245,8 +245,15 @@ def clip(gdf, mask, keep_geom_type=False):
...
-    # preserve the original order of the input
+    
+    # Return empty GeoDataFrame or GeoSeries if no shapes remain
+    if len(concat) == 0:
+        return (
+            GeoDataFrame(columns=gdf.columns, crs=gdf.crs)
+            if isinstance(gdf, GeoDataFrame)
+            else GeoSeries(crs=gdf.crs)
+        )
+    # Preserve the original order of the input
```
Comment (martinfleis) [inline geopandas/tools/clip.py:256]: Can you remove this commit from the PR?

File: geopandas/tools/tests/test_clip.py [hunk 2]
```diff
@@ -21,6 +21,21 @@ def point_gdf():
...
+    gdf = GeoDataFrame([Point(xy) for xy in pts], columns=["geometry"], crs="EPSG:4326")
+    return gdf
+
+@pytest.fixture
+def pointsoutside_overlap_gdf():
+    """Create a point GeoDataFrame. Its points are all outside the single
+    rectangle, and its bounds are overlapping the single rectangle's."""
+    pts = np.array([[5, 15], [15, 15], [15, 5]])
+    gdf = GeoDataFrame([Point(xy) for xy in pts], columns=["geometry"], crs="EPSG:4326")
+    return gdf
```
Comment (martinfleis) [inline geopandas/tools/tests/test_clip.py:38]: Remove this.

File: geopandas/tools/tests/test_clip.py [hunk 3]
```diff
@@ -315,6 +330,18 @@ def test_clip_line_keep_slivers(single_rectangle_gdf, sliver_line):
...
+    clipped = clip(pointsoutside_nooverlap_gdf, single_rectangle_gdf)
+    assert len(clipped) == 0
+
+
+def test_clip_box_overlap(pointsoutside_overlap_gdf, single_rectangle_gdf):
+    """Test clip when intersection is emtpy and boxes do overlap."""
+    clipped = clip(pointsoutside_overlap_gdf, single_rectangle_gdf)
+    assert len(clipped) == 0
+
+
```
Comment (martinfleis) [inline geopandas/tools/tests/test_clip.py:344]: Remove this.

**Comment by martinfleis** [comment]
Also, better to update you branch from master, that should fix CI.

**Comment by rwijtvliet** [comment]
Ahh yes of course, I meant to branch from master. Let me see if I can fix that and I'll do another PR

File: geopandas/tools/overlay.py [hunk 4]
```diff
@@ -139,16 +139,17 @@ def _overlay_union(df1, df2):
...
 
 def overlay(df1, df2, how="intersection", make_valid=True, keep_geom_type=True):
-    """Perform spatial overlay between two polygons.
+    """Perform spatial overlay between two GeoDataFrames.
 
-    Currently only supports data GeoDataFrames with polygons.
-    Implements several methods that are all effectively subsets of
-    the union.
+    Currently only supports data GeoDataFrames with uniform geometry types, 
+    i.e. containing only (Multi)Polygons, or only (Multi)Points, or a
```
Comment (jorisvandenbossche) [inline geopandas/tools/overlay.py:145]:
```suggestion
    Currently only supports data GeoDataFrames with uniform geometry types,
    i.e. containing only (Multi)Polygons, or only (Multi)Points, or a
```

**Comment by rwijtvliet** [comment]
Thanks Joris for the check. Is this what using Black would have alerted me about?

**Comment by martinfleis** [comment]
Thank you @rwijtvliet!

> Is this what using Black would have alerted me about?

In this case it would be `flake8`, not `black`. [We use both of these](https://geopandas.readthedocs.io/en/latest/contributing.html#style-guide-linting).

**Pull request merged**